### PR TITLE
dependencies: update `containerservice` to `2023-06-02-preview`

### PR DESCRIPTION
**Pull request by ms-henglu**

### CODE REVIEW

**Comment by tombuildsstuff** [comment]
Dependent on https://github.com/hashicorp/pandora/issues/3612

File: go.mod [hunk 1]
```diff
@@ -17,7 +17,7 @@ require (
 	github.com/google/go-cmp v0.5.9
 	github.com/google/uuid v1.3.1
 	github.com/hashicorp/go-azure-helpers v0.65.0
-	github.com/hashicorp/go-azure-sdk v0.20240111.1094251
+	github.com/hashicorp/go-azure-sdk v0.20240111.1120634
```
Comment (stephybun) [inline go.mod:20]: It's been mentioned frequently that the `go-azure-sdk` should always be bumped in a separate PR to avoid conflicts and the versions becoming outdated quickly, which has already happened in this case. Please remove this.

File: internal/services/containers/kubernetes_cluster_data_source.go [hunk 2]
```diff
@@ -1443,8 +1443,34 @@ func flattenKubernetesClusterDataSourceKubeConfigAAD(config kubernetes.KubeConfi
...
+	}
+	if input.PrincipalId != nil {
+		out.PrincipalId = *input.PrincipalId
+	}
+	if input.TenantId != nil {
+		out.TenantId = *input.TenantId
+	}
+
+	return identity.FlattenSystemOrUserAssignedMap(&out)
 }
```
Comment (stephybun) [inline internal/services/containers/kubernetes_cluster_data_source.go:1474]: This shouldn't be necessary anymore since a fix for this was pushed to Pandora
https://github.com/hashicorp/pandora/pull/3614
```suggestion
```

**Review by stephybun — CHANGES_REQUESTED** [review]

**Comment by tombuildsstuff** [comment]
hey @ms-henglu 

We’ve made some structural changes to `hashicorp/go-azure-sdk` which have split the SDK into multiple Go Modules. This means that the SDK which previously existed at `github.com/hashicorp/go-azure-sdk` now exists at:

* `github.com/hashicorp/go-azure-sdk/resource-manager` - for the Resource Manager SDK.
* `github.com/hashicorp/go-azure-sdk/sdk` - for the base layer.

We’ve updated the Provider to use the new Go Module structure in PR https://github.com/hashicorp/terraform-provider-azurerm/pull/24636 - as such would you mind rebasing this Pull Request on top of `main` to remove the merge conflict?

Thanks!

**Comment by ms-henglu** [comment]
Hi @tombuildsstuff , thanks! I've rebased this PR. Please take another look.

**Review by stephybun — APPROVED** [review]
Tests look fine, thanks @ms-henglu LGTM 👍
